### PR TITLE
Fix training_curve.py output

### LINKED ISSUE #915: Wrong output in training_curve.py
## Issue description

I’m testing the [training_curve.py](https://ivadomed.org/en/latest/scripts.html#ivadomed-training-curve) script and found an odd behavior. The script output a plot for each metrics from tensorboard event files. However, some plots don’t have the “right” content in them.

### Current behavior
For example, on rosenberg (master branch commit 93b9e73):
- `hausdorff_score.png` contains the tensorboard curve of `specificity`
- `intersection_over_union.png` contains the tensorboard curve of `precision`
- `precision_score.png` contains the tensorboard curve of `hausdorff`
- `specificity_score.png` contains the tensorboard curve of `intersection_over_union`

I also tried on my laptop and got different “bad” pairings (master branch commit 93b9e73):
- `accuracy_score.png` contains the tensorboard curve of `specificity`
- `dice_score.png` contains the tensorboard curve of `accuracy`
- `intersection_over_union.png` contains the tensorboard curve of `recall`
- `recall_score.png` contains the tensorboard curve of `dice`
- `precision_score.png` contains the tensorboard curve of `intersection_over_union`
- `specificity_score.png` contains the tensorboard curve of `precision`

### Steps to reproduce
- I’m using this command to plot curves from 2 models:
`ivadomed_training_curve -i path_output_model1,path_output_model2 -o path_output_for_png`
- Then I compre the png outputs with the tensorboard display.

Note: I have the same issue when using only one model path in `-i`.

### PR DESCRIPTION
**Pull request by mariehbourget**
#### GitHub

- [x] I've given this PR a concise, self-descriptive, and meaningful title
- [x] I've linked relevant issues in the PR body
- [x] I've applied [the relevant labels](https://www.neuro.polymtl.ca/software/contributing#pr_labels) to this PR
- [x] I've assigned a reviewer

#### PR contents

- [x] I've consulted [ivadomed's internal developer documentation](https://github.com/ivadomed/ivadomed/wiki) to ensure my contribution is in line with any relevant design decisions
- [ ] I've added [relevant tests](https://github.com/ivadomed/ivadomed/wiki/tests) for my contribution
- [x] I've updated the [relevant documentation](https://github.com/ivadomed/ivadomed/wiki/documentation) for my changes, including argparse descriptions, docstrings, and ReadTheDocs tutorial pages

## Description
This PR fixes #915 where there were mismatches between the PNG titles and the plot's content. This was due to the order of the `iterdir` listing (responsible for the order of `EventAccumulator`) being different from one machine to another, and different than the hardcoded metrics and losses lists order.

**Fixes:**
- The function `check_event_number` was renamed `get_events_path_list` and now returns a list of events path in alphabetical order. This order is then used to compile the lists of metrics and losses and used by the EventAccumulator (there are no hardcoded lists to maintain anymore).
- The `y_limits` were `[0, 1]` on each metrics plots. This was not suitable for `hausdorff score` with values ~10 (on my models). The `y_limits` are now automatically defined for the `hausdorff score`, the `y_limits` for others metrics remain `[0, 1]`.
- Updates/clarifications on comments and docstrings.

**New features:**
- In addition to plotting the training curves, the scripts now also saves data as a `csv` file for each model considered.
- A new option/argument `--lr` allows to plot the `learning_rate` in addition to losses and metrics. This is optional because:
    - learning_rate may not be necessary for the user and is likely to be the same across multiple trainings with the `--multiple` option.
    - learning_rate data is in the "main" events file at the root of the input_folder. This file can take ~2 minutes to load which could be cumbersome with option `--multiple` or several input_folders (`-i`).
- The `y_limits` of the `learning_rate` plot are automatically defined.
- For consistency, the `learning_rate` column in the `csv` file is appended at the end (last column), so the first 10 columns (metrics and losses) are always in the same order, with or without the `--lr` option.

## Linked issues
Fixes #915

### CODE REVIEW

**Review by charleygros — APPROVED** [review]
That was not an easy fix... Thanks for that @mariehbourget 
Tried a couple of config and works well, outputs are beaut'
Cheers
